### PR TITLE
dc/TACO-11-supreme adding zipkin tracing to Client and refactoring ca…

### PR DESCRIPTION
**Pull request by khappucino**
dc/TACO-11-supreme adding zipkin tracing to Client and refactoring call sites to support this as well as updating tests.  Incorporating PR comments from another branch wk/TACO-11-supreme

### CODE REVIEW

**Comment by khappucino** [comment]
Ok addressed the comments from https://github.com/xjdr/xio/pull/186

File: xio-core/src/main/java/com/xjeffrose/xio/http/ClientFactory.java [hunk 1]
```diff
@@ -3,26 +3,34 @@
...
+    this.tracing = tracing;
+  }
+
   public ClientChannelConfiguration channelConfig(ChannelHandlerContext ctx) {
     return ChannelConfiguration.clientConfig(ctx.channel().eventLoop());
   }
 
-  public abstract Client createClient(ChannelHandlerContext ctx, ClientConfig config);
+  public abstract Client createClient(
+      ChannelHandlerContext ctx, ClientConfig config, XioTracing tracing);
```
Comment (pdex) [inline xio-core/src/main/java/com/xjeffrose/xio/http/ClientFactory.java]: lose `tracing` here as a parameter and have the `@Override` method get at it through the getter

Comment (khappucino) [reply xio-core/src/main/java/com/xjeffrose/xio/http/ClientFactory.java]: fixed

File: xio-core/src/main/java/com/xjeffrose/xio/http/ClientFactory.java [hunk 2]
```diff
@@ -11,17 +11,23 @@
 /** Base class for generating http clients */
 public abstract class ClientFactory {
 
+  protected XioTracing tracing;
+
```
[the remark this answers is not in the document]
Comment (khappucino) [reply xio-core/src/main/java/com/xjeffrose/xio/http/ClientFactory.java]: ok so now it is 

@Getter private final XioTracing tracing;

fixed

**Pull request merged**